src/ciflow-bot.ts

```diff
@@ -0,0 +1,220 @@
+import * as probot from 'probot';
+
+export default function runCIFlowBot(app: probot.Application): void {
+  new CIFlowBot(app).main();
+}
+
+export class CIFlowBot {
+  // Constructor required
+  readonly app: probot.Application;
+
+  // Configurations
+  readonly allowed_command: string[] = ['ciflow'];
+  readonly bot_app_name = 'pytorchbot';
+  readonly bot_assignee = 'pytorchbot';
+  readonly pr_label_prefix = 'ciflow/';
+  readonly rollout_users = ['zhouzhuojie']; // slow rollout to specific group of users first
+
+  // Stateful instance variables
+  command = '';
+  command_args: string[] = [];
+  comment_author = '';
+  comment_body = '';
+  ctx: probot.Context = null;
+  dispatch_labels: string[] = [];
+  dispatch_strategies: string[] = ['add_default_labels'];
+  event = '';
+  owner = '';
+  pr_author = '';
+  pr_labels: string[] = [];
+  pr_number = 0;
+  repo = '';
+
+  constructor(app: probot.Application) {
+    this.app = app;
+  }
+
+  valid(): boolean {
+    if (this.event !== 'pull_request' && this.event !== 'issue_comment') {
```

Comment:
Can we have a GitHubConstants class that would contain all those string literals?